hrjobcontract/CRM/Hrjobcontract/Export/Converter.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+class CRM_Hrjobcontract_Export_Converter {
+
+  /**
+   * This method is based on CRM_Export_BAO_Export::writeCSVFromTable()
+   * but adds check for hrjobcontract entities and runs values converting
+   * on them so we get proper Job Contract entity values exported.
+   *
+   * @param $exportTempTable
+   * @param $headerRows
+   * @param $sqlColumns
+   * @param $exportMode
+   * @param null $saveFile
+   * @param string $batchItems
+   */
+  public static function writeCSVFromTable($exportTempTable, $headerRows, $sqlColumns, $exportMode, $saveFile = NULL, $batchItems = '') {
+    $converter = CRM_Hrjobcontract_ExportImportValuesConverter::singleton();
+    $componentDetails = array();
+
+    $dao = CRM_Core_DAO::executeQuery("SELECT * FROM {$exportTempTable}");
```

Comment:
this is the first solution came to my mind when I saw how civicrm implemented it .. but then I followed why they where doing the LIMIT thing and it was to keep memory usages limited and not to grow so much specially when the table contain a lot of rows  .. so I think implementing it  in the way I suggested may be a better approach .. what do you think ?